### PR TITLE
[Feature]: Add Leaflet Graph

### PR DESCRIPTION
**Pull request by nxanil**
@natabal 

Added `LeafLet Graph`. Please have a look.

leaflet

### CODE REVIEW

File: GraphsV2/LeafletGraph/index.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+const LeafletGraph = (props) => {
+
+  const [stateData, setStateData] = useState([]);
+  const [infowindow, setInfowindow] = useState({ data: null, position: null });
+
+  useEffect(() => {
+    initiate(props);
+
+    return () => {
```
Comment (natabal) [inline GraphsV2/LeafletGraph/index.js:55]: you should have a useEffect for this on unmounting.

File: GraphsV2/LeafletGraph/index.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+          <Data>&nbsp;</Data>
+          <Data>{row.text}</Data>
+        </InfoBox>
+      ))
+    }
+
+    return (
+      <InfoContainer className={'geoGraph'}>
+        {
+          displayInfo([
```
Comment (natabal) [inline GraphsV2/LeafletGraph/index.js:116]: This is specific to NSGs however I think this should be generic so we can add any objects with a location. We should define this in the vis config file similarly how we define the table columns. Let's take this opportunity to improve on the framework.

File: GraphsV2/LeafletGraph/index.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+      iconAnchor: labelOrigin,
+    });
+
+    return (
+      <Marker
+        key={data[idColumn]}
+        position={position}
+        icon={iconPerson}
+        onClick={() => handleMarkerClick(data, onMarkClick)}
+        onMouseOver={() => toggleInfoWindow(data, position)}
```
Comment (natabal) [inline GraphsV2/LeafletGraph/index.js:171]: if we define the tooltip info in the vis config maybe we add mouse events only if those are available. We might not have to do this for all the cases.

File: GraphsV2/LeafletGraph/index.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+      <SearchBar
+        data={data}
+        options={filters}
+        handleSearch={handleSearch}
+      />
+    );
+  }
+
+  const currentCenter = [
+    Number(process.env.REACT_APP_GOOGLE_MAP_LAT),
```
Comment (natabal) [inline GraphsV2/LeafletGraph/index.js:201]: we could rename these vars to remove Map API specifics and then re-use them.

**Comment by killanch** [comment]
@nxanil could you pls revive this PR?

**Comment by nxanil** [comment]
@killanch 

The PR is now ready for review. Please check once.

File: Graphs/LeafletGraph/index.js [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+              <img src='/icons/GREY.png'>
+                <span class="markerClusterLabel">${count}</span>
+              </img>
+          </div>`,
+      className: 'markerCluster',
+    });
+  }
+
+  const currentCenter = [
+    Number(process.env.REACT_APP_MAP_LAT),
```
Comment (killanch) [inline Graphs/LeafletGraph/index.js:295]: existing vars REACT_APP_GOOGLE_MAP_LAT and REACT_APP_GOOGLE_MAP_LNG are used in few other files within vis-graphs, could you pls refactor those to use the new vars?

**Review by killanch — COMMENTED** [review]
@nxanil could you pls attach gif with latest changes?

File: Graphs/LeafletGraph/index.js [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+    Number(process.env.REACT_APP_MAP_LNG),
+  ]
+
+  if (searchBar !== false) {
+    height -= 70;
+  }
+
+  return (
+    <React.Fragment>
+      {renderSearchBarIfNeeded()}
```
Comment (preshitg) [inline Graphs/LeafletGraph/index.js:305]: Minor: Can you rename these functions?
renderSearchBarIfNeeded -> renderSearchBar
renderMarkersIfNeeded -> renderMarkers
renderPolylineIfNeeded -> renderPolyline

**Review by killanch — APPROVED** [review]

**Review by preshitg — APPROVED** [review]

**Pull request merged**